developer/docs/arvo/generators.md

```diff
@@ -7,141 +7,149 @@ title: Using generators with apps
+  ^-  [(list move) _+>.$]                               ::  28
+  ~&  ping+message+(@t arg)                             ::  29
+  [~ +>.$]                                              ::  30
+::                                                      ::  31
+++  coup  |=(* `+>)                                     ::  32
+--                                                      ::  33
+```
 
 Now we can run this with:
 
-    ~fintud-macrep:dojo> |start %examples-ping
-    >=
-    ~fintud-macrep:dojo> :examples-ping &examples-ping-message [~sampel-sipnym 'heyya']
-    >=
-
-And on `~sampel-sipnym`, assuming it's running `:examples-ping` as well,
-we see `[%receiving 'heyya']`.
-
-This is an annoying way to invoke the app, though. The `[]` are
-mandatory, and optional arguments are hard. Let's make a generator,
-`+examples-send` to make everything nicer. Since it's specific to
-`:examples-ping`, let's put it in `/gen/examples/ping/send.hoon`:
-
-    :-  %say
-    |=  {^ {to/@p message/?($~ {text/@t $~})} $~}
-    [%examples-ping-message to ?~(message 'howdy' text.message)]
-
-A couple of new things here. Firstly, `message/?($~ {text/@t $~})`
-should be read as "the message is either null or a pair of text and
-null". Generator argument lists are always null-terminated, which makes
-it convenient to accept lists in tail position (which are particularly
-annoying without generators). `?(a b)` is the irregular form of
-`$?(a b)`, which is a union type. Thus, `?(a b)` means the type of
-anything that's in either `a` or `b`. Thus, in our case,
-`?(~ [text=@t ~])` is either null or a pair of text and null.
-
-Secondly, `?~(a b c)` is a rune which means "if the `a` is null, do `b`,
-else `c`". It is roughly equivalent to `?:(=(~ a) b c)`.
+```
+~fintud-macrep:dojo/examples> |start %ping
+>=
+~fintud-macrep:dojo/examples> :ping &ping-message [~sampel-sipnym 'heyya']
+>=
+```
+
+And on `~sampel-sipnym`, assuming it's running `:ping` as well, we see 
+`[%receiving 'heyya']`. You can test this yourself with a couple of comets.
+
+This is an annoying way to invoke the app, though. The `[]` are mandatory, 
+and optional arguments are hard. Let's make a generator, `+send` to make 
+everything nicer. Since it's specific to `:ping`, let's put it in 
+`/gen/ping/send.hoon`:
+
+```
+:-  %say
+|=  [^ {to=@p message=?($~ [text=@t $~])} $~}
+[%ping-message to ?~(message 'howdy' text.message)]
+```
+
+A couple of new things here. Firstly, `message=?($~ [text=@t $~])` should be 
+read as "the message is either null or a pair of text and null". Generator 
+argument lists are always null-terminated, which makes it convenient to accept 
+lists in tail position (which are particularly annoying without generators). 
+`?(a b)` is the irregular form of `$?(a b)`, which is a union type. Thus, 
+`?(a b)` means the type of anything that's in either `a` or `b`. Thus, in our 
+case, `?($~ {text/@t $~})` is either null or a pair of text and null.
```

Comment:
Also here.

Additionally, all symbols like `$foo` and `$~` can be switched to `%foo` and just `~`. If you need to manually switch from "value/noun mode" to "pattern/mold mode", you can use `,`, like `,%foo` and `,~`.